cli/lsp/documents.rs

```diff
@@ -2015,48 +2011,60 @@ impl Iterator for PreloadDocumentFinder {
       }
     }
 
-    while let Some(entry) = self.pending_entries.pop_front() {
-      match entry {
-        PendingEntry::SpecifiedRootFile(file) => {
-          // since it was a file that was specified as a root url, only
-          // verify that it's valid
-          if let Some(specifier) = Self::get_valid_specifier(&file) {
-            return Some(specifier);
+    // This first drains all the pending entries then adds the root dir entries
+    // one at a time to the pending entries before draining them. This is because
+    // we're traversing based on directory depth, so we want to search deeper
+    // directories first
+    while !self.pending_entries.is_empty() || !self.root_dir_entries.is_empty()
+    {
+      while let Some(entry) = self.pending_entries.pop_front() {
+        match entry {
+          PendingEntry::SpecifiedRootFile(file) => {
+            // since it was a file that was specified as a root url, only
+            // verify that it's valid
+            if let Some(specifier) = Self::get_valid_specifier(&file) {
```

Comment:
Do you expect this bit to have any impact on the LSP performance?

Reply:
There shouldn't be any change for the lsp unless someone had a lot of complex globs for different directories, in which case it will compare against less needless patterns now.

Reply:
Sounds good